C7/Game.cs

```diff
@@ -289,42 +296,23 @@ enum GameState {
 	}
 
 	public void _on_QuitButton_pressed() {
-		// This apparently exits the whole program
-		// GetTree().Quit();
-
 		// ChangeSceneToFile deletes the current scene and frees its memory, so this is quitting to main menu
 		GetTree().ChangeSceneToFile("res://MainMenu.tscn");
 	}
 
-	public void _on_Zoom_value_changed(float value) {
-		mapView.setCameraZoomFromMiddle(value);
-	}
-
 	public void AdjustZoomSlider(int numSteps, Vector2 zoomCenter) {
 		VSlider slider = GetNode<VSlider>("CanvasLayer/Control/SlideOutBar/VBoxContainer/Zoom");
 		double newScale = slider.Value + slider.Step * (double)numSteps;
-		if (newScale < slider.MinValue)
-			newScale = slider.MinValue;
-		else if (newScale > slider.MaxValue)
-			newScale = slider.MaxValue;
+		newScale = Mathf.Clamp(newScale, slider.MinValue, slider.MaxValue);
 
 		// Note we must set the camera zoom before setting the new slider value since setting the value will trigger the callback which will
 		// adjust the zoom around a center we don't want.
-		mapView.setCameraZoom((float)newScale, zoomCenter);
+		// camera.scaleZoom(zoom)
 		slider.Value = newScale;
 	}
 
-	public void _on_RightButton_pressed() {
-		mapView.cameraLocation += new Vector2(128, 0);
-	}
-	public void _on_LeftButton_pressed() {
-		mapView.cameraLocation += new Vector2(-128, 0);
-	}
-	public void _on_UpButton_pressed() {
-		mapView.cameraLocation += new Vector2(0, -64);
-	}
-	public void _on_DownButton_pressed() {
-		mapView.cameraLocation += new Vector2(0, 64);
+	private void onSliderZoomChanged(float value) {
```

Comment:
On my end it looks like zoom isn't working right now. If we either change the function here to _on_Zoom_value_changed, or if we change line 271 of C7Game.tscn to match this function, it starts working again. Might be a worthwhile fix. I'm not sure what naming convention we want to follow here.

Reply:
thanks, not sure when this broke. It should be named in camel case, not sure why a lot of these callbacks are _ prefixed, maybe when you create a signal in Godot this is/was the default naming convention is uses to generate the function